sql/src/test/java/io/crate/analyze/where/WhereClauseAnalyzerTest.java

```diff
@@ -253,248 +232,12 @@ public void testSelectWherePartitionedByColumn() throws Exception {
-        WhereClause whereClause = analyzeSelectWhere("select name from users where id='jalla'");
-
-        assertThat(whereClause.docKeys().get(), contains(isDocKey("jalla")));
-
-        whereClause = analyzeSelectWhere("select name from users where 'jalla'=id");
-        assertThat(whereClause.docKeys().get(), contains(isDocKey("jalla")));
-
-        whereClause = analyzeSelectWhere("select name from users where id='jalla' and id='jalla'");
-        assertThat(whereClause.docKeys().get(), contains(isDocKey("jalla")));
-
-        whereClause = analyzeSelectWhere("select name from users where id='jalla' and (id='jalla' or 1=1)");
-        assertThat(whereClause.docKeys().get(), contains(isDocKey("jalla")));
-
-        // since the id is unique it is not possible to have a result here, this is not optimized and just results in
-        // no found primary keys
-        whereClause = analyzeSelectWhere("select name from users where id='jalla' and id='kelle'");
-        assertFalse(whereClause.docKeys().isPresent());
-
-
-        whereClause = analyzeSelectWhere("select name from users where id='jalla' or name = 'something'");
-        assertFalse(whereClause.docKeys().isPresent());
-        assertFalse(whereClause.noMatch());
-
-        whereClause = analyzeSelectWhere("select name from users where name = 'something'");
-        assertFalse(whereClause.docKeys().isPresent());
-        assertFalse(whereClause.noMatch());
-
-    }
-
-    @Test
-    public void test4ColPrimaryKey() throws Exception {
-        WhereClause whereClause = analyzeSelectWhere(
-            "select * from pk4 where i1=10 and i2=20 and i3=30 and i4=40");
-        assertThat(whereClause.docKeys().get(), contains(isDocKey(10, 20, 30, 40)));
-        assertFalse(whereClause.noMatch());
-
-        whereClause = analyzeSelectWhere(
-            "select * from pk4 where i1=10 and i2=20 and i3=30 and i4=40 and i1=10");
-        assertThat(whereClause.docKeys().get(), contains(isDocKey(10, 20, 30, 40)));
-        assertFalse(whereClause.noMatch());
-
-
-        whereClause = analyzeSelectWhere("select * from pk4 where i1=1");
-        assertFalse(whereClause.docKeys().isPresent());
-        assertFalse(whereClause.noMatch());
-
-        whereClause = analyzeSelectWhere(
-            "select * from pk4 where i1=10 and i2=20 and i3=30 and i4=40 and i1=11");
-        assertFalse(whereClause.docKeys().isPresent());
-    }
-
-    @Test
-    public void test1ColPrimaryKeySetLiteralDiffMatches() throws Exception {
-        WhereClause whereClause = analyzeSelectWhere(
-            "select name from users where id in ('jalla', 'kelle') and id in ('jalla', 'something')");
-        assertFalse(whereClause.noMatch());
-        assertThat(whereClause.docKeys().get(), contains(isDocKey("jalla")));
-    }
-
-    @Test
-    public void test1ColPrimaryKeySetLiteral() throws Exception {
-        WhereClause whereClause = analyzeSelectWhere("select name from users where id in ('1', '2')");
```

Comment:
`any` cases are covered in `EqualityExtractorTest` - since we rewrite `IN` to `ANY` in the `ExpressionAnalyzer` I don't know what's special about these tests?

Reply:
In that case I'd be in favour of dropping them. We only need a single test that verifies that the integration between `WhereClauseOptimizer` and `EqualityExtractor` works. 

Edge-cases should be moved to the `EqualityExtractor` test if necessary